new(mc): linux telegraf pack (#3397)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/operatingsystems-linux-telegraf-agent.md

<Tabs groupId="sync">
<TabItem value="OS-Linux-Telegraf-Agent-custom" label="OS-Linux-Telegraf-Agent-custom">

| Service Alias | Service Template             | Service Description                                                                                                                                |
|:--------------|:-----------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------|
| Cpu           | OS-Linux-Cpu-Telegraf-Agent-custom    | Check the rate of utilization of CPUs for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPUs |
| Load          | OS-Linux-Load-Telegraf-Agent-custom   | Check the server load average                                                                                                                      |
| Memory        | OS-Linux-Memory-Telegraf-Agent-custom | Check the rate of the utilization of memory                                                                                                        |

> | Ntp           | OS-Linux-Ntp-Telegraf-Agent-custom    | Check system time synchronization with NTP server                                                                                                  |

Reviewer: ```suggestion
| Ntp           | OS-Linux-Ntp-Telegraf-Agent-custom    | Check system time synchronization with an NTP server                                                                                                  |
```

</TabItem>
<TabItem value="Systemd-Sc-Status" label="Systemd-Sc-Status">

| Metric name                    | Unit  |
|:-------------------------------|:------|
| systemd.services.running.count | count |
| systemd.services.failed.count  | count |
| systemd.services.dead.count    | count |
| systemd.services.exited.count  | count |
| *sc*#status                    | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Traffic" label="Traffic">

| Metric name                                     | Unit  |
|:------------------------------------------------|:------|
| *interface*#status                              | N/A   |
| *interface*#interface.traffic.in.bitspersecond  | b/s   |
| *interface*#interface.traffic.out.bitspersecond | b/s   |

</TabItem>
<TabItem value="Uptime" label="Uptime">

| Metric name           | Unit  |
|:----------------------|:------|
| system.uptime.seconds | s     |

</TabItem>
</Tabs>

## Prerequisites

The prerequisites below have to be applied to the Linux servers to be monitored.

### Network flow

Two TCP flows must be open from the host to the poller.

> | Source | Destination    | Protocole | Port | Purpose |

Reviewer: ```suggestion
| Source | Destination    | Protocol | Port | Purpose |
```

| ------ | -------------- | --------- | ---- | --- |

> | Hôte supervisé | Collecteur  | TCP       | 1443 | Access to Telegraf agent's configuration. |

Reviewer: ```suggestion
| Monitored host | Poller | TCP       | 1443 | Access to the Telegraf agent's configuration. |
```

> | Hôte supervisé | Collecteur  | TCP       | 4317 | OpenTelemetry data flow. |

Reviewer: ```suggestion
| Monitored host | Poller | TCP       | 4317 | OpenTelemetry data flow. |
```

### System prerequisites on the poller

> To be able to use the Telegraf agent, you must use a poller with at least version `24.04.2` of `centreon-engine`. The Telegraf agent will configure itself via a HTTPS request sent to Centreon Engine.

1. For this to work, you must first get a valid certificate or generate a self-signed one on the poller as detailed below.

> > In the next prerequisites, replace `${HOSTNAME}` with the poller's FQDN if they don't match.

Reviewer: ```suggestion
> In the command below, replace `${HOSTNAME}` with the poller's FQDN if they don't match.
```

> > If you set the access parameters to the configuration server by IP address instead of FQDN,

Reviewer: ```suggestion
> If you set an IP address in the access parameters to the configuration server instead of an FQDN,
```

> the Telegraf agent will refuse the certificate.

```bash
openssl req -new -subj "/CN=${HOSTNAME}" -addext "subjectAltName = DNS:${HOSTNAME}" -newkey rsa:2048 -sha256 -days 365 -nodes -x509 -keyout /etc/centreon-engine/conf-server.key -out /etc/centreon-engine/conf-server.crt
chown centreon-engine: /etc/centreon-engine/conf-*
```

> The `-days 365` option limits the certificate validity to one year. You may choose a longer or shorter duration according to your security/maintainance preferences.

> 2. Then define Engine's configuration server for Telegraf.

Author: ```suggestion
2. Then provide Engine with the connection information it needs to give to the Telegraf agent so that it can send information to Engine.
```

> 2. Then define Engine's configuration server for Telegraf.

Reviewer: ?

```bash
cat > /etc/centreon-engine/otl_server.json <<EOF
{
    "otel_server": {
        "host": "0.0.0.0",
        "port": 4317,
        "encryption": true,
        "certificate_path": "/etc/centreon-engine/conf-server.crt",
        "key_path": "/etc/centreon-engine/conf-server.key"
    },
    "max_length_grpc_log": 0,
    "telegraf_conf_server": {
        "http_server" : {
            "port": 1443,
            "encryption": true,
            "certificate_path": "/etc/centreon-engine/conf-server.crt",
            "key_path": "/etc/centreon-engine/conf-server.key"
        },
        "engine_otel_endpoint": "${HOSTNAME}:4317",
        "check_interval":60
    }
}
EOF
chown centreon-engine: /etc/centreon-engine/otl_server.json
```

### Configuration de Centreon Engine

> 1. In the **Configuration > Pollers > Engine configuration** menu, under the **Data** tab, add an entry to the Broker modules to load and enter the `/usr/lib64/centreon-engine/libopentelemetry.so /etc/centreon-engine/otl_server.json` directive. Save the form.

Reviewer: ```suggestion
1. In the **Configuration > Pollers > Engine configuration** menu, on the **Data** tab, add an entry to the Broker modules to load and enter the `/usr/lib64/centreon-engine/libopentelemetry.so /etc/centreon-engine/otl_server.json` directive. Save the form.
```

> 2. Export the poller's configuration ___with a restart___ of **Centreon Engine**.

Reviewer: ```suggestion
2. Export the poller's configuration, selecting the **Restart** option.
```

update-ca-trust
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
openssl s_client -connect mypoller.local:1443 2>/dev/null </dev/null |  sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' >  /etc/pki/ca-trust/source/anchors/mypoller.local.crt
update-ca-trust
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
openssl s_client -connect mypoller.local:1443 2>/dev/null </dev/null |  sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > /usr/local/share/ca-certificates/mypoller.local.crt
update-ca-certificates -f
```

</TabItem>
</Tabs>

Check the certificate's validity with this command:

```bash
curl https://mypoller.local:1443/engine
```

At this stage, the expected response is:

```html
<html><body>No host service found from get parameters</body></html>
```

2. Install the Telegraf agent and some dependencies.

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf -y install epel-release
dnf -y config-manager --set-enabled 'powertools'
```

> This part is an exerpt from [Telegraf's official documentation](https://docs.influxdata.com/telegraf/v1/install/?t=RedHat+%26amp%3B+CentOS).

Reviewer: ```suggestion
This part is an excerpt from [Telegraf's official documentation](https://docs.influxdata.com/telegraf/v1/install/?t=RedHat+%26amp%3B+CentOS).
```

```bash
cat > /etc/yum.repos.d/influxdb.repo <<'EOF'
[influxdb]
name = InfluxData Repository - Stable
baseurl = https://repos.influxdata.com/stable/$basearch/main
enabled = 1
gpgcheck = 1
gpgkey = https://repos.influxdata.com/influxdata-archive_compat.key
EOF

yum install -y telegraf
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf -y install epel-release
dnf -y config-manager --set-enabled 'crb'
```

> This part is an exerpt from [Telegraf's official documentation](https://docs.influxdata.com/telegraf/v1/install/?t=RedHat+%26amp%3B+CentOS).

Reviewer: ```suggestion
This part is an excerpt from [Telegraf's official documentation](https://docs.influxdata.com/telegraf/v1/install/?t=RedHat+%26amp%3B+CentOS).
```

```bash
cat > /etc/yum.repos.d/influxdb.repo <<'EOF'
[influxdb]
name = InfluxData Repository - Stable
baseurl = https://repos.influxdata.com/stable/$basearch/main
enabled = 1
gpgcheck = 1
gpgkey = https://repos.influxdata.com/influxdata-archive_compat.key
EOF

yum install -y telegraf
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

> This part is an exerpt from [Telegraf's official documentation](https://docs.influxdata.com/telegraf/v1/install/).

Reviewer: ```suggestion
This part is an excerpt from [Telegraf's official documentation](https://docs.influxdata.com/telegraf/v1/install/).
```

```bash
wget -q https://repos.influxdata.com/influxdata-archive_compat.key -O influxdata-archive_compat.key
echo '393e8779c89ac8d958f81f942f9ad7fb82a25e133faddaf92e15b16e6ac9ce4c influxdata-archive_compat.key' | sha256sum -c && cat influxdata-archive_compat.key | gpg --dearmor | tee /etc/apt/trusted.gpg.d/influxdata-archive_compat.gpg > /dev/null
echo 'deb [signed-by=/etc/apt/trusted.gpg.d/influxdata-archive_compat.gpg] https://repos.influxdata.com/debian stable main' | sudo tee /etc/apt/sources.list.d/influxdata.list
apt-get update
```

> Si des erreurs concernant le dépôt **Sury** apparaissent à ce stade, lancez `apt-key adv --fetch-keys 'https://packages.sury.org/php/apt.gpg' > /dev/null 2>&1` puis `apt-get update`.

Reviewer: FR

```bash
apt-get -y install telegraf
```

</TabItem>
</Tabs>

> 3. Set-up the Telegraf agent so that it retrieves its configuration from the poller.

Reviewer: ```suggestion
3. Set up the Telegraf agent so that it retrieves its configuration from the poller.
```

```bash
cat > /etc/default/telegraf <<EOF
TELEGRAF_OPTS='--config-url-watch-interval 120s --config=https://mypoller.local:1443/engine?host=$HOSTNAME'
EOF
systemctl restart telegraf
```

> > - Make sure you replaced `mypoller.local` with the poller's FQDN.

Reviewer: ```suggestion
> - Make sure you replace `mypoller.local` with the poller's FQDN.
```

> - If the name of the host to monitor is different from the `$HOSTNAME` variable's content, you'll have to change it inside the `/etc/default/telegraf` file.

4. To allow the Telegraf agent to run checks on **systemd-journal**, run these commands.

```bash
usermod -a -G systemd-journal telegraf
systemctl restart telegraf
```

> repere

Reviewer: ?

baseurl=https://packages.centreon.com/rpm-plugins/el9/testing/noarch/
enabled=0
gpgcheck=1
gpgkey=https://yum-gpg.centreon.com/RPM-GPG-KEY-CES
module_hotfixes=1

[centreon-plugins-unstable]
name=Centreon plugins repository. (UNSUPPORTED)
baseurl=https://packages.centreon.com/rpm-plugins/el9/unstable/$basearch/
enabled=0
gpgcheck=1
gpgkey=https://yum-gpg.centreon.com/RPM-GPG-KEY-CES
module_hotfixes=1

[centreon-plugins-unstable-noarch]
name=Centreon plugins repository. (UNSUPPORTED)
baseurl=https://packages.centreon.com/rpm-plugins/el9/unstable/noarch/
enabled=0
gpgcheck=1
gpgkey=https://yum-gpg.centreon.com/RPM-GPG-KEY-CES
module_hotfixes=1
EOF

dnf install -y centreon-plugin-Operatingsystems-Linux-Local.noarch
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
wget -O- https://apt-key.centreon.com | gpg --dearmor | tee /etc/apt/trusted.gpg.d/centreon.gpg > /dev/null 2>&1
echo "deb https://packages.centreon.com/apt-plugins-stable/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/centreon-plugins.list
apt-get update && apt-get -y install centreon-plugin-operatingsystems-linux-local
```

</TabItem>
</Tabs>

> 6. Restart the telegraf service.

Reviewer: ```suggestion
6. Restart the **telegraf** service.
```
